internal/cmd/asyncapi/command_import.go

```diff
@@ -0,0 +1,402 @@
+	Kafka KafkaBinding `yaml:"kafka"`
+}
+
+type ConfluentBrokerXConfigs struct {
+	SaslMechanisms   string `yaml:"sasl.mechanisms"`
+	SaslPassword     string `yaml:"sasl.password"`
+	SaslUsername     string `yaml:"sasl.username"`
+	SecurityProtocol string `yaml:"security.protocol"`
+}
+type ConfluentBroker struct {
+	Type     string                  `yaml:"type"`
+	XConfigs ConfluentBrokerXConfigs `yaml:"x-configs"`
+}
+
+type ConfluentSRXConfigs struct {
+	BasicAuthUserInfo string `yaml:"basic.auth.user.info"`
+}
+type ConfluentSchemaRegistry struct {
+	Type     string              `yaml:"type"`
+	XConfigs ConfluentSRXConfigs `yaml:"x-configs"`
+}
+
+func newImportCommand(prerunner pcmd.PreRunner) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "import spec.yaml",
+		Short:   "Adds topics, schemas and tags from the yaml file.",
+		Args:    cobra.ExactArgs(1),
+		Example: "confluent asyncapi import spec.yaml",
+	}
+	c := &command{AuthenticatedStateFlagCommand: pcmd.NewAuthenticatedStateFlagCommand(cmd, prerunner)}
+	c.RunE = c.parse
+	c.Flags().Bool("overwrite", false, "Overwrite existing topics and schemas.")
+	c.Flags().String("kafka-api-key", "", "Kafka cluster API key.")
+	c.Flags().String("schema-registry-api-key", "", "API key for Schema Registry.")
+	c.Flags().String("schema-registry-api-secret", "", "API secret for Schema Registry.")
+	return c.Command
+}
+
+func getFlagsImport(cmd *cobra.Command) (*flagsImport, error) {
+	overwrite, err := cmd.Flags().GetBool("overwrite")
+	if err != nil {
+		return nil, err
+	}
+	kafkaApiKey, err := cmd.Flags().GetString("kafka-api-key")
+	if err != nil {
+		return nil, err
+	}
+	schemaRegistryApiKey, err := cmd.Flags().GetString("schema-registry-api-key")
+	if err != nil {
+		return nil, err
+	}
+	schemaRegistryApiSecret, err := cmd.Flags().GetString("schema-registry-api-secret")
+	if err != nil {
+		return nil, err
+	}
+
+	flags := &flagsImport{
+		overwrite:               overwrite,
+		kafkaApiKey:             kafkaApiKey,
+		schemaRegistryApiKey:    schemaRegistryApiKey,
+		schemaRegistryApiSecret: schemaRegistryApiSecret,
+	}
+	return flags, nil
+}
+
+func (c *command) parse(cmd *cobra.Command, strings []string) error {
+	spec, err := fileToStruct()
+	if err != nil {
+		return err
+	}
+	// Get flags
+	flagsImp, err := getFlagsImport(cmd)
+	if err != nil {
+		return err
+	}
+	//Getting Kafka Cluster & SR
+	flags := createExportFlagsFromImportFlags(flagsImp)
+	details, err := c.getAccountDetails(&flags)
+	if err != nil {
+		return err
+	}
+	for topicName, topicDetails := range spec.Channels {
+		err = addChannelToCluster(details, spec, topicName, topicDetails.TopicBinding.Kafka, false, flagsImp.overwrite)
+		if err != nil {
+			if err == fmt.Errorf("topic is already present and overwrite flag is false. Moving to the next topic") {
+				log.CliLogger.Info(err)
+			} else {
+				log.CliLogger.Warn(err)
+			}
+		}
+	}
+	return nil
+}
+
+func fileToStruct() (*Spec, error) {
+	fileName := os.Args[3]
```

Comment:
You should not be getting args from `os.Args`, they are passed to the command function after being validated.